### PR TITLE
Optimize code for better performance and readability  🚀

### PR DESCRIPTION
**Pull request by abhint**

### CODE REVIEW

File: run_cmake.py [hunk 1]
```diff
@@ -1,71 +1,57 @@
...
-            windowsArch = arg
+            cmake_args.append('-DCMAKE_BUILD_TYPE=Debug')
         elif arg != 'force':
-            if arg.startswith('-G'):
-                explicitGenerator = True
-            cmake.append(arg)
+            cmake_args.append(arg)
+
     if sys.platform == 'win32':
-        if windowsArch == 'x64':
```
Comment (ilya-fedin) [inline run_cmake.py:26]: Have you just removed the possibility to build a x64 binary?

File: run_cmake.py [hunk 2]
```diff
@@ -1,71 +1,57 @@
...
             for line in f:
                 target = line.strip()
                 if len(target) > 0:
-                    cmake.append('-DDESKTOP_APP_SPECIAL_TARGET=' + target)
+                    cmake_args.append('-DDESKTOP_APP_SPECIAL_TARGET=' + target)
 
-    cmake.extend(['-Werror=dev', '-Werror=deprecated', '--warn-uninitialized', '..' if not buildType else '../..'])
-    command = '"' + '" "'.join(cmake) + '"'
+    cmake_args.extend(['-Werror=dev', '-Werror=deprecated', '--warn-uninitialized', '..' if not buildType else '../..'])
+    command = ' '.join(cmake + cmake_args)
```
Comment (ilya-fedin) [inline run_cmake.py:39]: It doesn't really feels like a right thing to do. This `cmake + cmake_args` concatenation has absolutely no sense and what's more important this removes the double quote guards for the arguments.

File: run_cmake.py [hunk 3]
```diff
@@ -1,71 +1,57 @@
...
-        if windowsArch == 'x64':
-            cmake.append('-Ax64')
-        else:
-            cmake.append('-AWin32') # default
-    elif windowsArch != '':
-        print("[ERROR] x86/x64 switch is supported only on Windows.")
-        return 1
+        cmake_args.append('-AWin32')
     elif sys.platform == 'darwin':
-        if not explicitGenerator:
```
Comment (ilya-fedin) [inline run_cmake.py:34]: Again, it looks like it removes a feature

File: run_cmake.py [hunk 4]
```diff
@@ -1,71 +1,57 @@
-# This file is part of Desktop App Toolkit,
-# a set of libraries for developing nice desktop applications.
-#
-# For license and copyright information please follow this link:
-# https://github.com/desktop-app/legal/blob/master/LEGAL
```
Comment (ilya-fedin) [inline run_cmake.py:5]: This is just wrong as well

File: run_cmake.py (hunk 2 above)
Comment (mymedia2) [reply run_cmake.py:39]:
```suggestion
    command = shlex.join(cmake + cmake_args)
```

**Pull request closed without merging**